tmt/steps/report/reportportal.py

```diff
@@ -38,113 +35,172 @@ class ReportReportPortalData(tmt.steps.report.ReportStepData):
-        else:
-            self.debug(f"Response code from the server: {response.status_code}")
-            self.debug(f"Message from the server: {message}")
-            self.info("report", "Successfully uploaded.", "yellow")
+
+            # Create a launch
+            self.info("launch", launch_name, color="green")
+            response = session.post(url=f"{url}/launch",
+                                    headers=headers,
+                                    json={"name": launch_name,
+                                          "description": self.step.plan.summary,
+                                          "attributes": attributes,
+                                          "startTime": timestamp()})
+            self.handle_response(response)
+            launch_uuid = tmt.utils.yaml_to_dict(response.text).get("id")
+            assert launch_uuid is not None
+            self.info("    uuid", launch_uuid, "yellow")
+
+            # For each test
+            for result in self.step.plan.execute.results():
+                test = [test for test in self.step.plan.discover.tests()
+                        if test.serialnumber == result.serialnumber][0]
+                if test.contact:
+                    attributes.append({"key": "contact", "value": test.contact[0]})
+                env_vars = [{'key': k, 'value': v} for k, v in test.environment.items()
+                            if not k.startswith("TMT_") or k in TMT_env_vars]
+
+                #  Create a test item
+                self.info("test", result.name, color="green")
+                response = session.post(url=f"{url}/item",
+                                        headers=headers,
+                                        json={"name": result.name,
+                                              "description": test.summary,
+                                              "attributes": attributes,
+                                              "parameters": env_vars,
+                                              "codeRef": test.web_link() or None,
+                                              "launchUuid": launch_uuid,
+                                              "type": "TEST",
+                                              "testCaseId": test.id or None,
+                                              "startTime": timestamp()})
+                self.handle_response(response)
+                item_uuid = tmt.utils.yaml_to_dict(response.text).get("id")
+                assert item_uuid is not None
+                self.info("    uuid", item_uuid, "yellow")
+
+                # For each log
+                for log_path in result.log:
+                    try:
+                        log = self.step.plan.execute.read(log_path)
+                    except (IndexError, AttributeError):
+                        log = None
+                    if not log:
+                        continue
```

Comment:
Can we short the code and `continue` directly in the `except` block?

Reply:
I tried, but it failed on mypy:
tmt/steps/report/reportportal.py:178: error: Unsupported operand types for + ("str" and "None")  [operator]
tmt/steps/report/reportportal.py:178: note: Right operand is of type "Optional[str]"
Found 1 error in 1 file (checked 53 source files)